src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +18,44 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add phone to DB. Phone: " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return phone;
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            Predicate findAllPredicate = criteriaBuilder.and();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                CriteriaBuilder.In<Object> in = criteriaBuilder.in(root.get(entry.getKey()));
+                for (String value : entry.getValue()) {
+                    in.value(value);
+                    findAllPredicate = criteriaBuilder.and(findAllPredicate, in);
+                }
```

Comment:
Варто винести цю операцію з циклу.
```suggestion
                }
                findAllPredicate = criteriaBuilder.and(findAllPredicate, in);
```